File: algorithm/力扣热题100/1_两数之和.js
// https://leetcode.cn/problems/two-sum/
function twoSum(nums, target) {
    const map = new Map()
    for (let i = 0, num; i < nums.length; i++) {
        num = nums[i];
        if (map.has(target - num)) {
            return [map.get(target - num), i]
        }
        map.set(num, i)
    }
    return []
}

File: algorithm/力扣热题100/1_两数之和.go
package main

func twoSum(nums []int, target int) []int {
	m := map[int]int{}
	for index, num := range nums {
		if index2, ok := m[target-num]; ok {
			return []int{index2, index}
		}
		m[num] = index
	}
	return nil
}
